yt_dlp/extractor/rai.py

```diff
@@ -35,139 +28,123 @@ class RaiBaseIE(InfoExtractor):
+            # remove \r\n\t before and after <![CDATA[ ]]> to avoid
+            # polluted text with xpath_text
+            s = re.sub(r'(\]\]>)[\r\n\t]+(</)', '\\1\\2', s)
+            return re.sub(r'(>)[\r\n\t]+(<!\[CDATA\[)', '\\1\\2', s)
+
         if not re.match(r'https?://', relinker_url):
             return {'formats': [{'url': relinker_url}]}
 
-        formats = []
-        geoprotection = None
-        is_live = None
-        duration = None
-
-        for platform in ('mon', 'flash', 'native'):
-            relinker = self._download_xml(
-                relinker_url, video_id,
-                note=f'Downloading XML metadata for platform {platform}',
-                transform_source=fix_xml_ampersands,
-                query={'output': 45, 'pl': platform},
-                headers=self.geo_verification_headers())
-
-            if xpath_text(relinker, './license_url', default='{}') != '{}':
-                self.report_drm(video_id)
-
-            if not geoprotection:
-                geoprotection = xpath_text(
-                    relinker, './geoprotection', default=None) == 'Y'
+        # set User-Agent to generic 'Rai' to avoid quality filtering from
+        # the media server and get the maximum qualities available
+        relinker = self._download_xml(
+            relinker_url, video_id, note='Downloading XML metadata',
+            transform_source=fix_cdata, query={'output': 64},
+            headers={**self.geo_verification_headers(), 'User-Agent': 'Rai'})
 
-            if not is_live:
-                is_live = xpath_text(
-                    relinker, './is_live', default=None) == 'Y'
-            if not duration:
-                duration = parse_duration(xpath_text(
-                    relinker, './duration', default=None))
+        if xpath_text(relinker, './license_url', default='{}') != '{}':
+            self.report_drm(video_id)
 
-            url_elem = find_xpath_attr(relinker, './url', 'type', 'content')
-            if url_elem is None:
-                continue
+        is_live = xpath_text(relinker, './is_live', default='N') == 'Y'
+        duration = parse_duration(xpath_text(relinker, './duration', default=None))
+        media_url = xpath_text(relinker, './url[@type="content"]', default=None)
 
-            media_url = url_elem.text
+        if not media_url:
+            self.raise_no_formats('The relinker returned no media url')
 
+        # geo flag is a bit unreliable and not properly set all the time
+        geoprotection = xpath_text(relinker, './geoprotection', default='N') == 'Y'
+        if '/video_no_available.mp4' in media_url:
             # This does not imply geo restriction (e.g.
             # http://www.raisport.rai.it/dl/raiSport/media/rassegna-stampa-04a9f4bd-b563-40cf-82a6-aad3529cb4a9.html)
-            if '/video_no_available.mp4' in media_url:
-                continue
+            self.report_warning('Video not available. Likely due to geo-restriction.')
```

Comment:
per discussion in #5672 it may be preferable to make this `raise_geo_restricted` instead

Reply:
fixed in next commit